### ISSUE TITLE
BrowserSync - Cannot read property 'toString' of undefined

### ISSUE THREAD

**Issue by regenrek** [BrowserSync/browser-sync#1254]
Hi,

I'm currently stuck on a problem with webpack + browsersync. Everytime I try to
start up my app it stops on the watch task.
I don't think it's a bug with browsersync but currently I'm not able to locate the issue. Does anybody else have this kind of error?

Regards!

### Issue details
```

3009
{ port: '3009',
  logLevel: 'debug',
  proxy: 
   { target: 'http://mvtb.dev',
     middleware: [ [Object], [Object] ] },
  files: [ 'templates/**/*.php', 'src/**/*.php' ] }
[BS] [debug] -> Starting Step: Finding an empty port
[BS] [debug] Found a free port: undefined
/srv/www/mvtb/web/app/themes/mjtheme/node_modules/browser-sync/lib/browser-sync.js:544
    bs.debug("Setting Option: {cyan:%s} - {magenta:%s", name, value.toString());

```
### Please specify which version of Browsersync, node and npm you're running

- Browsersync [ 2.18.2 ]
- Node        [  6.5.0  ]
- Npm         [  3.10.3  ]

### Affected platforms

- [ ] linux
- [ ] windows
- [X] OS X
- [ ] freebsd
- [ ] solaris
- [ ] other _(please specify which)_

### Browsersync use-case

- [] API
- [ ] Gulp
- [ ] Grunt
- [ ] CLI

### If CLI, please paste the entire command below

{cli command here}

### for all other use-cases, (gulp, grunt etc), please show us exactly how you're using Browsersync

{Browsersync init code here}

**Comment by regenrek** [comment]
EDIT: 

**TL;TR:** Fix:   `npm install portscanner@1.0.0  `

**Long Story:**
Found it. I got another project which is exactly identical where browsersync works without any problems. So I compared the _package.json_ file.

Then I saw I had two different browser-sync versions installed.

WORKING
`browser-sync@2.17.5`

NOT WORKING:
`browser-sync@2.18.2`

So I thought I would go back to 2.17.5 and everything works fine. 
Nope. Good then i diff compared npm-list and saw that portscanner was
missing in the npm tree (which is related to my error above).
Then npm install portscanner@1.0.0 fixed the issue.


Regards Kevin

**Comment by samkuhn** [comment]
I also had the same problem.. shouldn't we leave this issue open?
Is it that browser-sync has a missing dep?

**Comment by regenrek** [comment]
I reopen the issue as @samkuhn suggested

**Comment by strarsis** [comment]
Isn't this also an issue with browsersync option parsing?
Ideally browsersync would either complain about the unexpected port value type or even attempt to convert it.

**Comment by shakyShane** [comment]
Fixed in browser-sync@2.18.4

Thanks @regenrek for reporting :)
